runtime/integration-tests/src/integration_tests.rs

```diff
@@ -1882,3 +1882,276 @@ fn cdp_treasury_handles_honzon_surplus_correctly() {
+				Change::NewValue(Some(Ratio::saturating_from_rational(200, 100))),
+				Change::NewValue(1_000_000 * dollar(USD_CURRENCY)),
+			));
+			assert_ok!(Dex::add_liquidity(
+				Origin::signed(AccountId::from(BOB)),
+				RELAY_CHAIN_CURRENCY,
+				USD_CURRENCY,
+				5 * dollar(RELAY_CHAIN_CURRENCY),
+				10 * dollar(USD_CURRENCY),
+				0,
+				false,
+			));
+			// Alice has all Bob's permissions now
+			assert_ok!(Proxy::add_proxy(
+				Origin::signed(AccountId::from(BOB)),
+				AccountId::from(ALICE),
+				ProxyType::Any,
+				0
+			));
+			let root_call = Box::new(Call::Currencies(module_currencies::Call::update_balance(
+				AccountId::from(ALICE).into(),
+				NATIVE_CURRENCY,
+				1000 * dollar(NATIVE_CURRENCY) as i128,
+			)));
+			let gov_call = Box::new(Call::Tips(pallet_tips::Call::report_awesome(
+				b"bob is awesome".to_vec(),
+				AccountId::from(BOB),
+			)));
+			let transfer_call = Box::new(Call::Currencies(module_currencies::Call::transfer(
+				AccountId::from(BOB).into(),
+				NATIVE_CURRENCY,
+				10 * dollar(NATIVE_CURRENCY),
+			)));
+			let adjust_loan_call = Box::new(Call::Honzon(module_honzon::Call::adjust_loan(
+				RELAY_CHAIN_CURRENCY,
+				10 * dollar(RELAY_CHAIN_CURRENCY) as i128,
+				10 * dollar(USD_CURRENCY) as i128,
+			)));
+			let authorize_loan_call = Box::new(Call::Honzon(module_honzon::Call::authorize(
+				RELAY_CHAIN_CURRENCY,
+				AccountId::from(BOB).into(),
+			)));
+			let dex_swap_call = Box::new(Call::Dex(module_dex::Call::swap_with_exact_target(
+				vec![RELAY_CHAIN_CURRENCY, USD_CURRENCY],
+				dollar(USD_CURRENCY),
+				dollar(RELAY_CHAIN_CURRENCY),
+			)));
+			let dex_add_liquidity_call = Box::new(Call::Dex(module_dex::Call::add_liquidity(
+				RELAY_CHAIN_CURRENCY,
+				USD_CURRENCY,
+				10 * dollar(RELAY_CHAIN_CURRENCY),
+				10 * dollar(USD_CURRENCY),
+				0,
+				false,
+			)));
+
+			// Proxy calls do not bypass root permision
+			assert_ok!(Proxy::proxy(
+				Origin::signed(AccountId::from(ALICE)),
+				AccountId::from(BOB),
+				None,
+				root_call.clone()
+			));
+			// while the proxy call executes the call being proxied fails
+			assert_eq!(
+				Currencies::free_balance(NATIVE_CURRENCY, &AccountId::from(ALICE)),
+				100 * dollar(NATIVE_CURRENCY)
+			);
+
+			// Alice's gives governance permissions to Bob
+			assert_ok!(Proxy::add_proxy(
+				Origin::signed(AccountId::from(ALICE)),
+				AccountId::from(BOB),
+				ProxyType::Governance,
+				0
+			));
+			// Bob can be a proxy for alice gov call
+			assert_ok!(Proxy::proxy(
+				Origin::signed(AccountId::from(BOB)),
+				AccountId::from(ALICE),
+				None,
+				gov_call.clone()
+			));
+			let hash = BlakeTwo256::hash_of(&(BlakeTwo256::hash(b"bob is awesome"), AccountId::from(BOB)));
+			// last event was sucessful tip call
+			assert_eq!(
+				System::events()
+					.into_iter()
+					.map(|r| r.event)
+					.filter_map(|e| if let Event::Tips(inner) = e { Some(inner) } else { None })
```

Comment:
why not using `System::assert_last_event`?

Reply:
It doesn't really matter, but that is more explicit. The result of `System::assert_last_event` is just `pallet_proxy::Event::ProxyExecuted(Ok())` which is less specific, but I don't feel strongly either way

Reply:
So that verifies that the event from the tip pallet did indeed return what was expected